Framework/Algorithms/src/MonteCarloAbsorption.cpp

```diff
@@ -175,12 +178,12 @@ MonteCarloAbsorption::doSimulation(const MatrixWorkspace &inputWS,
   for (int64_t i = 0; i < nhists; ++i) {
     PARALLEL_START_INTERUPT_REGION
 
-    const auto &xvalues = outputWS->readX(i);
-    auto &signal = outputWS->dataY(i);
-    auto &errors = outputWS->dataE(i);
+    auto &xvalues = outputWS->x(i);
```

Comment:
It seems to be skipping bins though, maybe you should first figure out if this is performance relevant.